### PR TITLE
treewide: Fix appimageTools breakages, use pname instead of name

### LINKED ISSUE #267508: todoist-electron: pname seems wrong
### Describe the bug
todoist-electron is unfree, so I have `"todoist-electron"` in my `allowUnfreePredicate`. However, I recently cannot evaluate my configuration regardless:

```
$ home-manager --flake .#default -b old build
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'home-manager-generation'
         whose name attribute is located at /nix/store/0y446mjvwzrhv723jh0lpkaz9gw8jzbq-source/pkgs/stdenv/generic/make-derivation.nix:311:7

       … while evaluating attribute 'buildCommand' of derivation 'home-manager-generation'

         at /nix/store/0y446mjvwzrhv723jh0lpkaz9gw8jzbq-source/pkgs/build-support/trivial-builders/default.nix:87:14:

           86|       enableParallelBuilding = true;
           87|       inherit buildCommand name;
             |              ^
           88|       passAsFile = [ "buildCommand" ]

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Package ‘todoist-electron-8.9.3’ in /nix/store/0y446mjvwzrhv723jh0lpkaz9gw8jzbq-source/pkgs/applications/misc/todoist-electron/default.nix:40 has an unfree license (‘unfree’), refusing to evaluate.

       a) To temporarily allow unfree packages, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNFREE=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnfree = true; }
       in configuration.nix to override this.

       Alternatively you can configure a predicate to allow specific packages:
         { nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [
             "todoist-electron-8.9.3"
           ];
         }

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnfree = true; }
       to ~/.config/nixpkgs/config.nix.
```

From the fact it says I should include `"todoist-electron-8.9.3"` in my `allowUnfreePredicate`, I assume that #264333 has somehow caused the (p)name of the package to change. Or the newly-added appimage machinery is mangling things somehow?

### Steps To Reproduce
1. Run `nix run nixpkgs/nixos-unstable#todoist-electron`
2. See `error: Package ‘todoist-electron-8.9.3’ in /nix/store/0y446mjvwzrhv723jh0lpkaz9gw8jzbq-source/pkgs/applications/misc/todoist-electron/default.nix:40 has an unfree license (‘unfree’), refusing to evaluate.`

### Expected behavior
todoist-electron should already be allowed by my configuration, without having to hardcode the version number (or do shenanigans to allow any package matching `todoist-electron-*` or whatever).

### Notify maintainers

@kylesferrazza @pokon548

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.0, NixOS, 23.11 (Tapir), 23.11.20231104.85f1ba3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.1`
 - nixpkgs: `/etc/nixpkgs/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by pbsds**
## Description of changes

depends on and fixes #271071
I'll rebase and fix merge commits once it is merged

fixes https://github.com/NixOS/nixpkgs/issues/267508

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/golden-cheetah-bin/default.nix [hunk 1]
```diff
@@ -17,7 +17,6 @@ appimageTools.wrapType2 {
   extraPkgs = pkgs: with pkgs; [ R zlib libusb-compat-0_1 ];
 
   extraInstallCommands = ''
-    mv $out/bin/${pname}-${version} $out/bin/GoldenCheetah
```
Comment (eclairevoyant) [inline pkgs/applications/misc/golden-cheetah-bin/default.nix:20]:
```suggestion
    mv $out/bin/${pname} $out/bin/GoldenCheetah
```

File: pkgs/applications/science/biology/jbrowse/default.nix [hunk 2]
```diff
@@ -19,9 +19,6 @@ appimageTools.wrapType2 {
   unshareIpc = false;
 
   extraInstallCommands = ''
-    mkdir -p $out/bin
-    mv $out/bin/jbrowse-${version} $out/bin/jbrowse-desktop
```
Comment (eclairevoyant) [inline pkgs/applications/science/biology/jbrowse/default.nix:23]:
```suggestion
    mv $out/bin/jbrowse $out/bin/jbrowse-desktop
```

File: pkgs/applications/video/lbry/default.nix [hunk 3]
```diff
@@ -23,9 +21,6 @@ in appimageTools.wrapAppImage rec {
 
   # General fixup
   extraInstallCommands = ''
-    # Firstly, rename the executable to lbry for convinence
-    mv $out/bin/${name} $out/bin/lbry
```
Comment (eclairevoyant) [inline pkgs/applications/video/lbry/default.nix:27]:
```suggestion
    mv $out/bin/${pname} $out/bin/lbry
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
caught a few changes to fix (double checked against the desktop files)

**Comment by jbgi** [comment]
Rebased version of this pr in #283222.

**Comment by pbsds** [comment]
These fixes has been moved back into https://github.com/NixOS/nixpkgs/pull/271071.

thanks for your review @eclairevoyant, it's been addressed in https://github.com/NixOS/nixpkgs/pull/271071 :)

**Pull request closed without merging**